### PR TITLE
Change listener to be plugable

### PR DESCRIPTION
**Pull request by loverend**
I have changed the listener to be plugable in the same way that the
publisher transport is.

I have also added a kafka listener too.

### CODE REVIEW

File: napalm_logs/listener/kafka.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    def start(self):
+        '''
+        Start listening for messages  
+        '''
+        # Start suicide polling thread
+        signal.signal(signal.SIGTERM, self._exit_gracefully)
+        self.__up = True
+        try:
+       	    self.consumer = kafka.KafkaConsumer(bootstrap_servers='{}:{}'.format(self.address, self.port), group_id='napalm-logs')
```
Comment (mirceaulinic) [inline napalm_logs/listener/kafka.py]: This indentation doesn't look fine. Can you check?
Also, when passing multiple kwargs, please separate them on new lines.

File: napalm_logs/listener/kafka.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        except kafka.errors.NoBrokersAvailable as err:
+            log.error(err, exc_info=True)
+            raise NapalmLogsException(err)
```
Comment (mirceaulinic) [inline napalm_logs/listener/kafka.py]: Maybe we should introduce a new exception class for listener.

File: napalm_logs/listener/tcp.py [hunk 3]
```diff
@@ -79,10 +66,12 @@ def _tcp_connection(self, conn, addr):
         log.debug('Closing connection with {}'.format(addr))
         conn.close()
 
-    def start_tcp(self):
+    def start(self):
         '''
-        Start listening for messages on TCP and queue them
+        Start listening for messages
         '''
+	signal.signal(signal.SIGTERM, self._exit_gracefully)
```
Comment (mirceaulinic) [inline napalm_logs/listener/tcp.py]: Looks like unidented block.

File: napalm_logs/listener/udp.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            skt = socket.socket(socket.AF_INET6, socket_type)
+        else:
+            skt = socket.socket(socket.AF_INET, socket_type)
+        return skt
+
+    def start(self):
+        '''
+        Start listening for messages
+        '''
+	signal.signal(signal.SIGTERM, self._exit_gracefully)
```
Comment (mirceaulinic) [inline napalm_logs/listener/udp.py]: Same here

File: napalm_logs/listener/udp.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        self.__up = True
+        self.skt = self._open_socket(socket.SOCK_DGRAM)
+        try:
+            self.skt.bind((self.address, self.port))
+        except socket.error as msg:
+            error_string = 'Unable to bind to port {} on {}: {}'.format(self.port, self.address, msg)
+            log.error(error_string, exc_info=True)
+            raise BindException(error_string)
```
Comment (mirceaulinic) [inline napalm_logs/listener/udp.py]: If we introduce a new exception class, don't forger to raise it here too.

File: napalm_logs/transport/base.py [hunk 6]
```diff
@@ -14,8 +14,5 @@ def __init__(self, addr, port):
     def start(self):
         pass
 
-    def publish(self, obj):
```
Comment (mirceaulinic) [inline napalm_logs/transport/base.py]: Why do we remove this?

Comment (loverend) [inline napalm_logs/transport/base.py]: Added back, this was unintended

**Pull request merged**